service/operator/operatorkit.go

```diff
@@ -0,0 +1,146 @@
+package operator
```

Comment:
This is what I would like to put into the operatorkit repo. I see the code in this file as base structure for all operator related logic. In case you align your operator implementation to the interface and concept ideas of this file, it is way more easy to get the reconciliation right. Or at least more right than wrong. At the moment our operators are dumb CRUD services. We do only create and delete once. We do not really reconcile and do not really fix existing but broken state. The interface defined and documented here aims to help with getting the idea of a specific operator implementation right from the beginning.